js/bruh-dash.js

```diff
@@ -10,89 +10,215 @@ var global = window || GLOBAL;
+      return array;
+    }else if (typeof amount !== "number"){
+      array.splice(1, array.length);
+      return array;
+    }else{
+      return array.slice(0, amount);
+    }
   },
 
   // creates a slice of an array with n elements taken from the end
-  takeRight: function () {
-
+ /* takeRight: function (array, amount) {
+    if (amount === 0){
+      var emptyArray = [];
+      return emptyArray;
+    }else if (amount !== "number"){
+      array.splice(0, array.length-1);
+      console.log(array);
+      return array;
+    }else if (amount > array.length){
+      return array;
+    }else{
+       array.splice(0, value-1);
+       console.log(array);
+       return array;
+    }
+  },*/
+    takeRight: function (array,value) {
+    if(typeof value !== "number"){
+      array.splice(0, array.length-1);
+      //console.log(array);
+      return array;
+    }else if(value === 0){
+      array.splice(0, array.length);
+      return array;
+    }else if(value>array.length){
+      return array;
+    }else{
+      array.splice(0, value-1);
+      //console.log(array);
+      return array;
+    }
   },
 
+
   // fills elements of array with specified value from the start index
   // up to but not including the end index
-  fill: function() {
-
+  fill: function(array, element, start, end) {
+    for (var i = 0; i < array.length; i++){
+      if (!start || !end){
+        array[i] = element;
+        console.log(array);
+      }else{
+        for (var j = start; j < end; j++){
+          array[j] = element;
+          console.log(array);
+        }
+      }
+    }
+    return array;
   },
 
   // removes all given values from an array
-  pull: function () {
-
+  pull: function (array1, array2) {
+      for (var i = 0; i < array1.length; i++){
+        for (var j = 0; j < array2.length; j++){
+          if (array2[j] === array1[i]){
+            array1.splice(i,1);
+            console.log(array1);
+          }
+        }
+      }
+      return array1;
   },
 
   // removes elements of an array corresponding to the given indices
-  pullAt: function () {
-
+  pullAt: function (array1, array2) {
+    var count = 0;
+    for (var i = 0; i < array1.length; i++){
+      for (var j = 0; j < array2.length; j++){
+        if (array2[j] === i){
+          console.log(array1.splice(array2[j]-count, 1));
+          count++;
+        }
+      }
+    }
+    return array1;
   },
 
-  // creates an array excluding all the specified values
-  without: function() {
 
+
+  // creates an array excluding all the specified values
+  without: function(array1, array2) {
+      for (var i = 0; i < array1.length; i++){
+        for (var j = 0; j < array2.length; j++){
+          if (array2[j] === array1[i]){
+            array1.splice(i,1);
+            console.log(array1);
+          }
+        }
+      }
+      return array1;
   },
 
   // returns an array with specified values excluded
-  difference: function() {
-
+  difference: function(array1, array2) {
```

Comment:
This is not working correctly.
given `difference([1,2,3], [1,2,5,1,2])` the result should be `[3,5]`